### PR TITLE
full register/login/auth backend functionality of our app

### PR DESCRIPTION
**Pull request by nard1n**
This PR includes the full functioning backend of the login/authorization of our app. This is linked to an Atlas db on MongoDB. 

This will allow our users to register, log in to access certain pages accessible only to logged-in users, and log out 

This has been successfully tested with Postman to register and login

### CODE REVIEW

File: routes/api/users.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            // Create JWT Payload
+            const payload = {
+                id: user.id,
+                username: user.username
+            };
+            
+            // Sign token
+            jwt.sign(
+                payload,
+                //keys.secretOrKey,
```
Comment (sossw1) [inline routes/api/users.js]: This may be why the JWT token is returning undefined, undo commenting

File: server.js [hunk 2]
```diff
@@ -1,40 +1,47 @@
...
+const express = require('express');
+const mongoose = require('mongoose');
+const cors = require('cors');
+const passport = require('passport');
+const users = require('./routes/api/users');
 
 //Express Setup
 const app = express();
-app.use(express.static (__dirname));
+//app.use(express.static (__dirname));
```
Comment (sossw1) [inline server.js]: Unused code line 11

File: server.js [hunk 3]
```diff
@@ -1,40 +1,47 @@
...
 
-const expressSession = require('express-session')({
-    secret: 'secret',
-    resave: false,
-    saveUninitialized: false
-});
+// const expressSession = require('express-session')({
```
Comment (sossw1) [inline server.js:13]: Need this for sessions

Comment (sossw1) [inline server.js]: Unused code lines 13-17

File: server.js [hunk 4]
```diff
@@ -1,40 +1,47 @@
...
 app.use(express.json({ limit: "30mb", extended: true }));
-app.use(express.urlencoded({ extended: false}));
-app.use(expressSession);
-app.use(cors());
-
-//passport setup
-app.use(passport.initialize());
-app.use(passport.session());
+app.use(express.urlencoded({ extended: false }));
+//app.use(expressSession);
```
Comment (sossw1) [inline server.js]: Unused code lines 21-22

File: server.js [hunk 5]
```diff
@@ -1,40 +1,47 @@
...
+mongoose.connect(db, { useNewUrlParser: true, useUnifiedTopology: true})
+    .then (() => console.log("MongoDB successfully connected"))
+    .catch ((error) => console.log(error.message));
 
 mongoose.set('useFindAndModify', false);
 
-const port = process.env.PORT || 5000;
+//passport middleware setup
+app.use(passport.initialize());
+//app.use(passport.session());
```
Comment (sossw1) [inline server.js]: Unused code line 36

File: routes/api/users.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+const keys = require ('../../config/keys');
+
+//load input validation 
+const validateRegisterInput = require ('../../validation/register');
+const validateLoginInput = require ('../../validation/login');
+
+//load user model
+const User = require ('../../models/User');
+
+// @route POST api/users/register
```
Comment (sossw1) [inline routes/api/users.js:14]: Unused code lines 14-16

File: routes/api/users.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+              .save()
+              .then(user => res.json(user))
+              .catch(err => console.log(err));
+          });
+        });
+      }
+    });
+  });
+
+// @route POST api/users/login
```
Comment (sossw1) [inline routes/api/users.js:52]: Unused code lines 52-54

**Review by sossw1 — COMMENTED** [review]
Line 87 of routes/api/users.js the keys config is commented out, which may have made JWT return an undefined token.

Single comments in some files show places where unused code was commented instead of deleted.

**Comment by nard1n** [comment]
making updates according to suggestions

**Review by sossw1 — APPROVED** [review]
No unused code left and the token generates properly, nice work!

**Pull request merged**